src/web/db/task_store_pymysql.py

```
@@ -0,0 +1,475 @@
+            # Check for an existing active task
+            row = fetch_query(
+                """
+                SELECT * FROM tasks
+                WHERE normalized_title = %s AND status NOT IN (%s, %s)
+                ORDER BY created_at DESC
+                LIMIT 1
+                """,
+                [normalized_title, *TERMINAL_STATUSES],
+            )
+            if row:
+                raise TaskAlreadyExistsError(self._row_to_task(row[0]))
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)